comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/cli/Main.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2020 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.outputcomparator.cli;
+
+import com.beust.jcommander.JCommander;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import org.mobilitydata.gtfsvalidator.outputcomparator.util.ValidationReport;
+
+public class Main {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final String REFERENCE_JSON = "report.json";
+  private static final String LATEST_JSON = "latest.json";
+  private static final String INTEGRATION_REPORT_JSON = "integration_report.json";
+  private static final Gson GSON = new Gson();
+
+  public static void main(String[] argv) {
+    ComparatorArguments args = new ComparatorArguments();
+    new JCommander(args).parse(argv);
+    File[] outputDirectory = new File(args.getOutputBase()).listFiles();
+    if (outputDirectory == null) {
+      logger.atSevere().log("Empty output directory");
+      return;
+    }
+    ImmutableMap.Builder<String, Object> mapBuilder = new Builder<>();
+    float badDatasetCount = 0;
```

Comment:
What was the reason to use float instead of int for a counter?

In general, please use double instead of float.

If an approximate value is what you want, the type to choose first is double. The problem with float is that it is very imprecise -- you can only trust about six digits of it -- and this causes real bugs. double is good to about fifteen digits, and is also better supported by common libraries such as streams.

Reply:
It should have been an `int` - probably a bad copy/paste. Thanks for the clarification around float vs doubles!